code/__HELPERS/unsorted.dm

```diff
@@ -1105,46 +1108,54 @@ var/global/image/action_purple_power_up
 	for(var/i in 1 to numticks)
 		sleep(delayfraction)
 		time_remaining -= delayfraction
-		if(!istype(L) || has_target && !istype(target)) // Checks if L exists and is not dead and if the target exists and is not destroyed
+		if(show_busy_icon)
+			if(has_busy_icon && user.alpha < 255)
+				busy_user.overlays += busy_icon
+				busy_icon.alpha = user.alpha
+				has_busy_icon = TRUE
+			else if(!has_busy_icon && user.alpha == 255)
+				busy_user.overlays += busy_icon
+				has_busy_icon = TRUE
+		if(!istype(busy_user) || has_target && !istype(target)) // Checks if busy_user exists and is not dead and if the target exists and is not destroyed
```

Comment:
sanity check should come before the code block using busy_user I think

Reply:
What's a sanity check? Is that 
> if(!istype(busy_user) || has_target && !istype(target)) // Checks if busy_user exists and is not dead and if the target exists and is not destroyed

Reply:
Nevermind I see